documentation/modules/exploit/linux/local/exim4_deliver_message_priv_esc.md

```diff
@@ -0,0 +1,85 @@
+# Vulnerable Application
+
+Exim 4.87 - 4.91 Local Privilege Escalation
+
+This module implements a flaw found in Exim versions 4.87 to 4.91 (inclusive). Improper validation of recipient address in deliver_message() function in /src/deliver.c may lead to remote command execution (CVE-2019-10149).
+
+Both meterpreter shell and classic shell are supported. The exploit will try to upload the payload you give him, set the suid bit on this payload and execute it to create a new root session. In order for the new session to be a root one, both PrependSetuid and PrependSetgid must be set to true (which is the default configuration for the exploit).
+
+# Creating A Testing Environment
+
+You basically just need to have a exim (between 4.87 and 4.91 inclusive) running and listening on a port (port 25 by default).
+For my tests, I used a VM with Ubuntu 18.04 LTS and exim 4.89 (I tested all the versions from 4.87 to 4.91). The exim source code can be downloaded from the official website (all the old versions can be found).
+You can also use this good Docker image which sets up a container with a vulnerable exim version running (https://github.com/dhn/exploits/tree/master/CVE-2019-10149).
+
+Before using the exploit, make sure exim is actually listening on a port (it may sound stupid, but I struggled a bit when creating a testing environment). However, you should not have any problem if you use the Docker image linked above.
+
+# Verification Steps
+
+1. `use exploit/linux/local/exim4_deliver_message_priv_esc`
+2. `set SESSION [session]`
+3. `set PAYLOAD [payload]`
+4. `set LHOST [lhost]`
+5. `set LPORT [lport]`
+6. `exploit`
+
+# Options
+
+## PAYLOAD
+
+Set this option to choose which type of root session you want to create (obviously you should try a meterpreter payload first).
```

Comment:
`(obviously you should try a meterpreter payload first).`

Disputable and presumptuous. What if I want to pull code from Rapid9?